### DISCUSSION TITLE
Welcome to Sqitch Discussions!

### DISCUSSION THREAD

**Discussion by theory** [theory/sqitch#607] [Announcements]
## 👋 Welcome!

The Sqitch community warmly invites your participation. We welcome discussion and contributions from anyone and everyone, at every level of knowledge and skill. In the interest of fostering an open and welcoming environment, all members of the community must abide by the [Sqitch Community Code of Conduct](http://sqitch.org/coc/). We hope that you:

*   Ask questions you’re wondering about
*   Share ideas, techniques, and knowledge
*   Engage with other community members
*   Welcome others and are open-minded. Remember that this is a community we build together 💪

We appreciate your participation, kindness, and generosity to others!

**Comment by mememia** [comment]
Hey,

i'm a student and I am an intern , i start with sqitch the company has two project,the first is a data base produced by sqitch and the second is the playlists ok !
 i wanna integrated this two project and i don't know how i can start ,have any idea about that please?
Thank you !

**Reply by theory** [reply]
You can dump the second database to a file, and make that file the first change in a new Sqitch project.

**Reply by mememia** [reply]
Hey,can you explain more please 👍

**Reply by theory** [reply]
Not without more information about what you have and what you're trying to do. But most database engines have an export to SQL function. In Postgres, for example, you can use `pg_dump --schema-only` to write to a file. Then use that file as the first change in your new Sqitch project. This allows you to start with where the database is today, and only worry about making future changes with Sqitch.

**Comment by R-Whitmer** [comment]
Is there any reasonably-short path to SQL Server support?

I have a great need for dependency-driven change management in SQL. I have created custom SQL scripts that do this sort of thing, but not without pain in writing changes that have to invoke framework functions. Having it handled externally like sqitch is the better way.

But my last few days trying to apply sqitch was doomed:

1. Windows poorly supported whereas many devs of databases use windows, as little as I like it.
2. Perl makes it very hard for anyone to install anywhere
3. Lack of any support for SQL Server, which is where our data is. Wishes for postgres cannot make it otherwise.

I've spent most of a week only to find out that previous SQL Server support work has been eradicated, all AI hallucinations to the contrary notwithstanding.

I guess that puts me back the a track of writing my own tools again. Too bad.

**Reply by theory** [reply]
> 3. Lack of any support for SQL Server, which is where our data is. Wishes for postgres cannot make it otherwise.

Thanks, this made me LOL.

Discussion has been in [this issue](https://github.com/sqitchers/sqitch/issues/149). As [I last said there](https://github.com/sqitchers/sqitch/issues/149#issuecomment-2236549632):

> I'd still like to see it happen, but someone will have to decide to do it.

I don't suppose you have the bandwidth to work on it rather than invent your own thing. [Strawberry Perl](https://strawberryperl.com) works well on windows, and the [OS workflow](https://github.com/sqitchers/sqitch/actions/workflows/os.yml) ensures that all the basics work on Windows, though only the SQLite engine is tested there. Still, though, I would think that once the pattern for installation was established, following the precedents of the engines that use ODBC drivers to make an mssql.pm would be a relatively straightforward exercise, assuming a standard mssql CLI of some kind. Not trivial, mind, probably a few weeks work, but then it'd be done.

**Reply by R-Whitmer** [reply]
I have worked for companies where open source was the model but cannot get permission at present to even submit bug fixes, whch is not a good position for me to be in. The powers that be often fail to see the value of what I do. I have bandwidth and will work nights and weeks if I have to, but I am subject to powers that be.

AIs I consulted gave me the impression first that sql server support for sqitch was released, then when they couldn't tell me how to make it work that it was unreleased, then when I downloaded from github and it still wasn't there, they said it was in a former branch that had been deleted. I have no idea where the truth lies. It doesn't seem to me that it should be too hard to do, so the idea that it might have been deleted makes me sad.

I can probably put together a good approximation of similar functionality that feel is needed that supports SQL Server, (still a bigger user base than Postgres) in maybe 40 hours, but I would simplify by writing it in C++ against the ODBC, using some great variadic ODBC templates I have created, bypassing sqlcmd or any abstraction or any functions that are not immediately important from my use case. It is statically linked which makes installation nearly as simple as copying a binary. I have no doubt you believe perl makes it simpler, but I have a different opinion and the fact remains that it somehow seems difficult for it to support SQL Server.

The sqitch model is similar to schema version control I implemented previously. I would seldom use it in anything but dependency driven mode. If I could make it work, there is nothing else out there that is like it that I can find. Certainy not the popular offerings.
 But doing my own (yet again) will feel better than continuing to beat my head against the wall trying to use sqitch.
 
The documentation that states "Strawberry Perl works well on Windows" is short and I have trouble reconciling it with the experience I had trying to make it work on Windows 11, and after a while, the AIs concurred, however simple solving the actual failure should have been. There was no documentation. When I follow the instructions and cannot find a sqitch executable anywhere on the system to run, and am not a perl developer so I don't know what to expect, I finally decided that devs would have to deploy it under WSL and Ubuntu, which I had more success with.

But lack of sql server support after all the false info from the AIs was the final show stopper.

**Reply by theory** [reply]
AIs are untrustworthy. You're better off asking here.

But lets' move discussion to #149, yes?

**Reply by R-Whitmer** [reply]
Yes and yes. I should have looked a little harder.
